framework/src/utils/SlepcSupport.C

```diff
@@ -0,0 +1,242 @@
+/****************************************************************/
+/*               DO NOT MODIFY THIS HEADER                      */
+/* MOOSE - Multiphysics Object Oriented Simulation Environment  */
+/*                                                              */
+/*           (c) 2010 Battelle Energy Alliance, LLC             */
+/*                   ALL RIGHTS RESERVED                        */
+/*                                                              */
+/*          Prepared by Battelle Energy Alliance, LLC           */
+/*            Under Contract No. DE-AC07-05ID14517              */
+/*            With the U. S. Department of Energy               */
+/*                                                              */
+/*            See COPYRIGHT for full restrictions               */
+/****************************************************************/
+
+#ifndef SLEPCSUPPORT_H
+#define SLEPCSUPPORT_H
+
+#include "libmesh/libmesh_config.h"
+
+#ifdef LIBMESH_HAVE_SLEPC
+
+#include "SlepcSupport.h"
+// MOOSE includes
+#include "MultiMooseEnum.h"
+#include "InputParameters.h"
+#include "EigenProblem.h"
+#include "Conversion.h"
+
+namespace Moose
+{
+namespace SlepcSupport
+{
+
+  InputParameters
+  getSlepcValidParams()
+  {
+    InputParameters params = emptyInputParameters();
+
+    MooseEnum eigen_solve_type("POWER ARNOLDI KRYLOVSCHUR JACOBI_DAVIDSON");
+    params.addParam<MooseEnum>("eigen_solve_type", eigen_solve_type,
+                               "POWER: Power / Inverse / RQI "
+                               "ARNOLDI: Arnoldi "
+                               "KRYLOVSCHUR: Krylov-Schur "
+                               "JACOBI_DAVIDSON: Jacobi-Davidson ");
+   return params;
+  }
+
+  InputParameters
+  getSlepcEigenProblemValidParams()
+  {
+    InputParameters params = emptyInputParameters();
+
+    // We are solving a Non-Hermitian eigenvalue problem by default
+    MooseEnum eigen_problem_type("HERMITIAN NON_HERMITIAN GEN_HERMITIAN GEN_NON_HERMITIAN GEN_INDEFINITE POS_GEN_NON_HERMITIAN", "NON_HERMITIAN");
+    params.addParam<MooseEnum>("eigen_problem_type", eigen_problem_type,
+                               "HERMITIAN: Hermitian "
+                               "NON_HERMITIAN: Non-Hermitian "
+                               "GEN_HERMITIAN: Gerneralized Hermitian "
+                               "GEN_NON_HERMITIAN: Generalized Non-Hermitian "
+                               "GEN_INDEFINITE: Generalized indefinite Hermitian "
+                               "POS_GEN_NON_HERMITIAN: Generalized Non-Hermitian with positive (semi-)definite B");
```

Comment:
Doco string should _also_ mention something like "Type of the eigenvalue problem we are solving", then it can elaborate on what the values are...